src/terms.jl

```diff
@@ -543,18 +543,48 @@ vectorize(x) = [x]
     coefnames(term::AbstractTerm)
 
 Return the name(s) of column(s) generated by a term.  Return value is either a
-`String` or an iterable of `String`s.
+`Symbol` or an iterable of `String`s.
 """
+StatsBase.coefnames(t::Term) = t.sym
 StatsBase.coefnames(t::FormulaTerm) = (coefnames(t.lhs), coefnames(t.rhs))
-StatsBase.coefnames(::InterceptTerm{H}) where {H} = H ? "(Intercept)" : []
-StatsBase.coefnames(t::ContinuousTerm) = string(t.sym)
-StatsBase.coefnames(t::CategoricalTerm) = 
-    ["$(t.sym): $name" for name in t.contrasts.termnames]
-StatsBase.coefnames(t::FunctionTerm) = string(t.exorig)
-StatsBase.coefnames(ts::TupleTerm) = reduce(vcat, coefnames.(ts))
+StatsBase.coefnames(::InterceptTerm{H}) where {H} = H ? Symbol(:Intercept) : []  # this seems like the wrong thing to return
+StatsBase.coefnames(t::ContinuousTerm) = t.sym
+StatsBase.coefnames(t::CategoricalTerm) = [Symbol("$(t.sym): $name") for name in t.contrasts.termnames]
+StatsBase.coefnames(t::FunctionTerm) = Symbol(string(t.exorig))
 StatsBase.coefnames(t::MatrixTerm) = mapreduce(coefnames, vcat, t.terms)
+#function StatsBase.coefnames(t::InteractionTerm)
+#    Symbol.(kron_insideout((args...) -> join(args, " & "), vectorize.(coefnames.(t.terms))...))
+#end
 StatsBase.coefnames(t::InteractionTerm) =
-    kron_insideout((args...) -> join(args, " & "), vectorize.(coefnames.(t.terms))...)
+    Symbol.(kron_insideout((args...) -> join(args, " & "), vectorize.(coefnames.(t.terms))...))
+StatsBase.coefnames(ts::TupleTerm) = _coefnames(ts.terms)
+_coefnames(ts::Tuple) = (coefnames(first(ts)), _coefnames(tail(ts))...)
+_coefnames(ts::Tuple{}) = ()
+
+"""
+    coef(term::AbstractTerm, s::Symbol)
+"""
+function StatsBase.coef(f::FormulaTerm, s::Symbol)
```

Comment:
I don't see what the point of this is, or rather why it's called `coef` (which means something different in statsbase).  can you explain what your goal is here?  in any case, this needs tests if it's going to be included, and it seems like it would be better in a separate PR.

also, it doesn't seem like this works for categorical terms or for interactions, even if you specify the full coefficient name, since `coefnames` will be a vector of symbols.

and the RHS can have other things than just a `MatrixTerm` (e.g., MixedModels has a tuple of fixed and random effect matrices which are of different types)